opencog/atoms/base/Atom.h

```diff
@@ -173,6 +173,11 @@ class Atom
 
     virtual ~Atom();
 
+    virtual const std::string& getName() const {
+        static std::string empty_string("");
+        return empty_string;
```

Comment:
My comment crossed yours...

In C++ `try..catch` is very fast. Python has much higher overhead.